examples/cpp/dds/RequestReplyExample/CalculatorServer.cpp

```diff
@@ -0,0 +1,224 @@
+
+        if (nullptr == publisher_)
+        {
+            return false;
+        }
+
+        subscriber_ = participant_->create_subscriber(eprosima::fastdds::dds::SUBSCRIBER_QOS_DEFAULT);
+
+        if (nullptr == subscriber_)
+        {
+            return false;
+        }
+
+        request_topic_ = participant_->create_topic("CalculatorRequest",
+                        request_type_.get_type_name(), eprosima::fastdds::dds::TOPIC_QOS_DEFAULT);
+
+        if (nullptr == request_topic_)
+        {
+            return false;
+        }
+
+        reply_topic_ = participant_->create_topic("CalculatorReply",
+                        reply_type_.get_type_name(), eprosima::fastdds::dds::TOPIC_QOS_DEFAULT);
+
+        if (nullptr == reply_topic_)
+        {
+            return false;
+        }
+
+        eprosima::fastdds::dds::DataWriterQos writer_qos;
+        writer_qos.history().kind = eprosima::fastdds::dds::KEEP_ALL_HISTORY_QOS;
+        reply_writer_ = publisher_->create_datawriter(reply_topic_, writer_qos);
+
+        if (nullptr == reply_writer_)
+        {
+            return false;
+        }
+
+        listener_ = { reply_writer_};
+
+        eprosima::fastdds::dds::DataReaderQos reader_qos;
+        reader_qos.reliability().kind = eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS;
+        reader_qos.durability().kind = eprosima::fastdds::dds::TRANSIENT_LOCAL_DURABILITY_QOS;
+        reader_qos.history().kind = eprosima::fastdds::dds::KEEP_ALL_HISTORY_QOS;
+        request_reader_ = subscriber_->create_datareader(request_topic_, reader_qos, &listener_);
+
+        if (nullptr == request_reader_)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    void deinit()
+    {
+        if (nullptr != request_reader_)
+        {
+            subscriber_->delete_datareader(request_reader_);
+            request_reader_ = nullptr;
+        }
+
+        if (nullptr != reply_writer_)
+        {
+            publisher_->delete_datawriter(reply_writer_);
+            reply_writer_ = nullptr;
+        }
+
+        if (nullptr != reply_topic_)
+        {
+            participant_->delete_topic(reply_topic_);
+            reply_topic_ = nullptr;
+        }
+
+        if (nullptr != request_topic_)
+        {
+            participant_->delete_topic(request_topic_);
+            request_topic_ = nullptr;
+        }
+
+        if ( nullptr != subscriber_)
+        {
+            participant_->delete_subscriber(subscriber_);
+            subscriber_ = nullptr;
+        }
+
+        if (nullptr != publisher_)
+        {
+            participant_->delete_publisher(publisher_);
+            publisher_ = nullptr;
+        }
+
+        if (nullptr != participant_)
+        {
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant_);
+            participant_ = nullptr;
+        }
```

Comment:
I suggest simplifying the code as it is an example, calling `DomainParticipant::delete_contained_entities` instead.

Reply:
Done in 0ab0ab5b5